### PR TITLE
[otbn] P-384 scalar multiplication

### PR DESCRIPTION
**Pull request by felixmiller**
This adds routine and test for P-384 scalar point multiplication in projective space.
This also reorgs the P-384 specific files moving everything to a single `p384.s` file, including the P-384 optimized Barret routine.

This is related to #5337 and #2982

### CODE REVIEW

**Review by imphil — APPROVED** [review]

File: sw/otbn/code-snippets/p384.s [hunk 1]
```diff
@@ -0,0 +1,1085 @@
...
+  bn.add    w16, w29, w2
+  bn.addc   w17, w30, w3
+  bn.sub    w10, w16, w12
+  bn.subb   w11, w17, w13
+  bn.sel    w16, w16, w10, C
+  bn.sel    w17, w17, w11, C
+  bn.mov    w29, w16
+  bn.mov    w30, w17
+
+ret
```
Comment (mcy) [inline sw/otbn/code-snippets/p384.s:712]: I think the indentation ran away from you here. =)

**Review by mcy — APPROVED** [review]
Looks good other than a little indentation nit.

File: sw/otbn/code-snippets/p384.s [hunk 2]
```diff
@@ -710,3 +710,376 @@ proj_add_p384:
...
+ *
+ * @param[in]  x20: dptr_x, pointer to dmem location containing affine
+ *                          x-coordinate of input point
+ * @param[in]  x21: dptr_y, pointer to dmem location containing affine
+ *                          y-coordinate of input point
+ * @param[in]  [w15, w14]: u[383:0] lower 384 bit of Barrett constant u for
+ *                                    modulus p
+ * @param[in]  [w13, w12]: p, modulus of P-384 underlying finite field
+ * @param[in]  w31: all-zero
+ * @param[in]  x18: dptr_P, pointer to dmem location to store resulting point
```
[the remark this answers is not in the document]
Comment (felixmiller) [reply sw/otbn/code-snippets/p384.s:738]: Yeah that's a bit annoying. Problem is that lower case p is usually the modulus of the finite field in literature (for ECC). I changed this now to `dptr_p_p`, `dptr_q_p`, `dptr_p_a`, `dptr_q_a`, to indicate that it is a curve point in projective or affine space respectively.

**Comment by felixmiller** [comment]
thanks @imphil and @mcy

**Pull request merged**